cargo install's "intermediate artifacts can be found at" message on failure is incorrect if a custom `build.build-dir` is configured

### Problem

When `cargo install` fails to compile the requested binary, it prints the following message:

```
error: could not compile `[package name]` (bin "[binary name]")
[…]
error: failed to compile `[package name] v[package version] ([package source path if `--path` provided])`, intermediate artifacts can be found at `[target directory path]`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

This suggestion is incorrect in the case when a custom `build.build-dir` path – distinct from the target directory path – is configured, either in a .cargo/config.toml file (local or global) or using the `CARGO_BUILD_BUILD_DIR` environment variable. The substance of the suggestion and the use of the term “intermediate artifacts” indicate that it's the `build.build-dir` path and `CARGO_BUILD_BUILD_DIR` environment variable that should be printed here.


### Steps

```sh
mkdir -p issue_16622/src
cd issue_16622
cat > Cargo.toml << 'EOF'
[package]
name = "issue-16622"
EOF
touch src/main.rs
CARGO_BUILD_BUILD_DIR=./build cargo install --path .
```

Output:
```
  Installing issue-16622 v0.0.0 ([…]/issue_16622)
   Compiling issue-16622 v0.0.0 ([…]/issue_16622)
error[E0601]: `main` function not found in crate `issue_16622`
  |
  = note: consider adding a `main` function to `src/main.rs`

For more information about this error, try `rustc --explain E0601`.
error: could not compile `issue-16622` (bin "issue-16622") due to 1 previous error
error: failed to compile `issue-16622 v0.0.0 ([…]/issue_16622)`, intermediate artifacts can be found at `[…]/issue_16622/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

### Possible Solution(s)

Desired output:
```
  Installing issue-16622 v0.0.0 ([…]/issue_16622)
   Compiling issue-16622 v0.0.0 ([…]/issue_16622)
error[E0601]: `main` function not found in crate `issue_16622`
  |
  = note: consider adding a `main` function to `src/main.rs`

For more information about this error, try `rustc --explain E0601`.
error: could not compile `issue-16622` (bin "issue-16622") due to 1 previous error
error: failed to compile `issue-16622 v0.0.0 ([…]/issue_16622)`, intermediate artifacts can be found at `[…]/issue_16622/build`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
```

### Notes

_No response_

### Version

```text
cargo 1.95.0-nightly (fe2f314ae 2026-01-30)
release: 1.95.0-nightly
commit-hash: fe2f314aef06e688a9517da1ac0577bb1854d01f
commit-date: 2026-01-30
host: aarch64-apple-darwin
libgit2: 1.9.2 (sys:0.20.3 vendored)
libcurl: 8.7.1 (sys:0.4.83+curl-8.15.0 system ssl:(SecureTransport) LibreSSL/3.3.6)
ssl: OpenSSL 3.5.4 30 Sep 2025
os: Mac OS 26.2.0 [64-bit]
```

--- a/src/cargo/ops/cargo_install.rs
+++ b/src/cargo/ops/cargo_install.rs
@@ -356,9 +356,9 @@ impl<'gctx> InstallablePackage<'gctx> {
                 "failed to compile `{}`, intermediate artifacts can be \
                  found at `{}`.\nTo reuse those artifacts with a future \
                  compilation, set the environment variable \
-                 `CARGO_TARGET_DIR` to that path.",
+                 `CARGO_BUILD_BUILD_DIR` to that path.",
                 self.pkg,
-                self.ws.target_dir().display()
+                self.ws.build_dir().display()
             )
         })?;
         let mut binaries: Vec<(&str, &Path)> = compile

--- a/tests/testsuite/directory.rs
+++ b/tests/testsuite/directory.rs
@@ -193,7 +193,7 @@ fn simple_install_fail() {
         .with_stderr_data(str![[r#"
 [INSTALLING] bar v0.1.0
 [ERROR] failed to compile `bar v0.1.0`, intermediate artifacts can be found at `[..]`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 Caused by:
   no matching package found
@@ -771,7 +771,7 @@ fn version_missing() {
         .with_stderr_data(str![[r#"
 [INSTALLING] bar v0.1.0
 [ERROR] failed to compile [..], intermediate artifacts can be found at `[..]`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 Caused by:
   failed to select a version for the requirement `foo = "^2"`

--- a/tests/testsuite/install.rs
+++ b/tests/testsuite/install.rs
@@ -1170,7 +1170,7 @@ fn compile_failure() {
 ...
 [ERROR] could not compile `foo` (bin "foo") due to 1 previous error
 [ERROR] failed to compile `foo v0.0.1 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/foo/target`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 ...
 "#]])
         .run();
@@ -2571,7 +2571,7 @@ fn failed_install_retains_temp_directory() {
 
 [ERROR] could not compile `foo` (bin "foo") due to 1 previous error
 [ERROR] failed to compile `foo v0.0.1`, intermediate artifacts can be found at `[..]`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 "#]]);
 
@@ -2620,8 +2620,8 @@ fn failed_install_points_to_build_dir_for_intermediate_artifacts() {
   | ^ expected one of `!` or `::`
 
 [ERROR] could not compile `foo` (bin "foo") due to 1 previous error
-[ERROR] failed to compile `foo v0.0.1 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/foo/target`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+[ERROR] failed to compile `foo v0.0.1 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/home/.cargo/build-artifacts`.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 "#]]);
 }
@@ -3014,7 +3014,7 @@ fn dry_run_incompatible_package_dependency() {
 [INSTALLING] foo v0.1.0 ([ROOT]/foo)
 [LOCKING] 1 package to latest compatible version
 [ERROR] failed to compile `foo v0.1.0 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/foo/target`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 Caused by:
   rustc [..] is not supported by the following package:

--- a/tests/testsuite/required_features.rs
+++ b/tests/testsuite/required_features.rs
@@ -766,7 +766,7 @@ Consider enabling some of the needed features by passing, e.g., `--features="a"`
         .with_stderr_data(str![[r#"
 [INSTALLING] foo v0.0.1 ([ROOT]/foo)
 [ERROR] failed to compile `foo v0.0.1 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/foo/target`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 Caused by:
   target `foo` in package `foo` requires the features: `a`
@@ -785,7 +785,7 @@ Caused by:
         .with_stderr_data(str![[r#"
 [INSTALLING] foo v0.0.1 ([ROOT]/foo)
 [ERROR] failed to compile `foo v0.0.1 ([ROOT]/foo)`, intermediate artifacts can be found at `[ROOT]/foo/target`.
-To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
+To reuse those artifacts with a future compilation, set the environment variable `CARGO_BUILD_BUILD_DIR` to that path.
 
 Caused by:
   target `foo` in package `foo` requires the features: `a`

